instrumentation/opentelemetry-instrumentation-wsgi/src/opentelemetry/instrumentation/wsgi/__init__.py

```diff
@@ -208,6 +215,46 @@ def collect_request_attributes(environ):
     return result
 
 
+def add_custom_request_headers(span, environ):
+    """Adds custom HTTP request headers into the span which are configured by the user
+    from the PEP3333-conforming WSGI environ to be used as span creation attributes as described
+    in the specification https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-request-and-response-headers"""
+    if span.kind == trace.SpanKind.SERVER:
+        attributes = {}
+        custom_request_headers_name = get_custom_headers(
+            OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SERVER_REQUEST
+        )
+        for header_name in custom_request_headers_name:
+            wsgi_env_var = header_name.upper().replace("-", "_")
+            header_values = environ.get(f"HTTP_{wsgi_env_var}")
+            if header_values:
+                key = normalise_request_header_name(header_name)
+                attributes[key] = [header_values]
+        span.set_attributes(attributes)
+
+
+def add_custom_response_headers(span, response_headers):
+    """Adds custom HTTP response headers into the sapn which are configured by the user from the
+    PEP3333-conforming WSGI environ as described in the specification
+    https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-request-and-response-headers"""
+    if span.kind == trace.SpanKind.SERVER:
```

Comment:
This check should be outside of the add_custom_response_headers() and in the calling function before you call add_custom_response_headers(). 
i.e. 
if span.kind == SERVER:
            add_custom_response_headers()

And inside the function it should just add it to span attributes

Reply:
for this use case you are right. but normally if there is an if condition which covers the whole function then that if condition should be outside of the function where that function is getting called, to make the function loosely coupled with that condition

Reply:
Okay. Will make the required changes.

Reply:
Changes done.